[Bug]: error TS7022: 'headingValue' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.

### Reproduction

when building the code using vite (npm run build)

### Describe the bug

when building code in vite I get the following error.
```
<script lang="ts" setup>
import { cn } from '@/lib/utils'
import { CalendarHeading, type CalendarHeadingProps, useForwardProps } from 'radix-vue'
import { computed, type HTMLAttributes } from 'vue'

const props = defineProps<CalendarHeadingProps & { class?: HTMLAttributes['class'] }>()

const delegatedProps = computed(() => {
  const { class: _, ...delegated } = props

  return delegated
})

const forwardedProps = useForwardProps(delegatedProps)
</script>

<template>
  <CalendarHeading
    v-slot="{ headingValue }"
    :class="cn('text-sm font-medium', props.class)"
    v-bind="forwardedProps"
  >
    <slot :heading-value>
      {{ headingValue }}
    </slot>
  </CalendarHeading>
</template>

```

### System Info

```bash
src/components/ui/calendar/CalendarHeading.vue:19:15 - error TS7022: 'headingValue' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.

19     v-slot="{ headingValue }"
                 ~~~~~~~~~~~~


Found 1 error in src/components/ui/calendar/CalendarHeading.vue:19
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

--- a/apps/www/src/lib/registry/default/ui/calendar/CalendarHeading.vue
+++ b/apps/www/src/lib/registry/default/ui/calendar/CalendarHeading.vue
@@ -5,6 +5,10 @@ import { computed, type HTMLAttributes } from 'vue'
 
 const props = defineProps<CalendarHeadingProps & { class?: HTMLAttributes['class'] }>()
 
+defineSlots<{
+  default: (props: { headingValue: string }) => any
+}>()
+
 const delegatedProps = computed(() => {
   const { class: _, ...delegated } = props
 

--- a/apps/www/src/lib/registry/default/ui/range-calendar/RangeCalendarHeading.vue
+++ b/apps/www/src/lib/registry/default/ui/range-calendar/RangeCalendarHeading.vue
@@ -5,6 +5,10 @@ import { computed, type HTMLAttributes } from 'vue'
 
 const props = defineProps<RangeCalendarHeadingProps & { class?: HTMLAttributes['class'] }>()
 
+defineSlots<{
+  default: (props: { headingValue: string }) => any
+}>()
+
 const delegatedProps = computed(() => {
   const { class: _, ...delegated } = props
 

--- a/apps/www/src/lib/registry/new-york/ui/calendar/CalendarHeading.vue
+++ b/apps/www/src/lib/registry/new-york/ui/calendar/CalendarHeading.vue
@@ -5,6 +5,10 @@ import { computed, type HTMLAttributes } from 'vue'
 
 const props = defineProps<CalendarHeadingProps & { class?: HTMLAttributes['class'] }>()
 
+defineSlots<{
+  default: (props: { headingValue: string }) => any
+}>()
+
 const delegatedProps = computed(() => {
   const { class: _, ...delegated } = props
 

--- a/apps/www/src/lib/registry/new-york/ui/range-calendar/RangeCalendarHeading.vue
+++ b/apps/www/src/lib/registry/new-york/ui/range-calendar/RangeCalendarHeading.vue
@@ -5,6 +5,10 @@ import { computed, type HTMLAttributes } from 'vue'
 
 const props = defineProps<RangeCalendarHeadingProps & { class?: HTMLAttributes['class'] }>()
 
+defineSlots<{
+  default: (props: { headingValue: string }) => any
+}>()
+
 const delegatedProps = computed(() => {
   const { class: _, ...delegated } = props
 

